packages/flutter/lib/src/services/binding.dart

```diff
@@ -20,15 +23,27 @@ mixin ServicesBinding on BindingBase {
   void initInstances() {
     super.initInstances();
     _instance = this;
-    window
-      ..onPlatformMessage = defaultBinaryMessenger.handlePlatformMessage;
+    window..onPlatformMessage = defaultBinaryMessenger.handlePlatformMessage;
     initLicenses();
   }
 
   /// The current [ServicesBinding], if one has been created.
   static ServicesBinding get instance => _instance;
   static ServicesBinding _instance;
 
+  /// The default [BinaryMessenger] instance.
+  @protected
+  BinaryMessenger defaultBinaryMessenger = const _DefaultBinaryMessenger._();
+
+  /// The default instance of [BinaryMessenger].
+  ///
+  /// This is used to send messages from the application to the platform, and
+  /// keeps track of which handlers have been registered on each channel so
+  /// it may dispatch incoming messages to the registered handler.
+  BinaryMessenger provideBinaryMessenger() {
+    return defaultBinaryMessenger;
+  }
+
```

Comment:
I think you want to flip these around. The getter should be the way to access this, to make it clear that we're not initializing a new value each time. The method should be protected, and should be called during initInstances to configure a field that defaultBinaryMessenger returns.

Reply:
Updated. I was wrong by treating the `binaryMessenger` field as private member (which is public accessible), and had this provideBinaryMessenger as the getter for it.